### ISSUE TITLE
Freezing execution

### ISSUE THREAD

**Issue by yoyecros** [ONLYOFFICE/CommunityServer#255]
Hi,

Docker install, latest version, all four servers (Community, Document, Mysql and Mail), I can't start the CommunityServer.

Here is the logs:
```
+ mkdir -p /var/www/onlyoffice/Data/Index/v7.4.0
++ ls -alhd /var/www/onlyoffice/Data/Index/v7.4.0
++ awk '{ print $3 }'
+ '[' elasticsearch '!=' elasticsearch ']'
+ chown -R elasticsearch:elasticsearch /var/log/onlyoffice//Index
+ dpkg -l
+ grep -q onlyoffice-xmppserver
+ sed '/web\.talk/s/value=\"\S*\"/value=\"true\"/g' -i /var/www/onlyoffice/WebStudio/web.appsettings.config
+ sed '/web\.chat/s/value=\"\S*\"/value=\"true\"/g' -i /var/www/onlyoffice/WebStudio/web.appsettings.config
+ systemctl stop onlyofficeRadicale
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeTelegram
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeSocketIO
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeThumb
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeFeed
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeIndex
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeJabber
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeMailAggregator
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeMailWatchdog
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeMailCleaner
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeNotify
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeBackup
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeStorageMigrate
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeStorageEncryption
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop onlyofficeUrlShortener
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop god
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl enable god
+ systemctl stop elasticsearch
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop redis-server
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop mysql
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop nginx
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl stop monoserveApiSystem.service
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl enable monoserveApiSystem.service
++ seq 1 1
+ for serverID in $(seq 1 ${APP_MONOSERVE_COUNT})
+ index=1
+ '[' 1 == 1 ']'
+ index=
+ systemctl stop monoserve.service
System has not been booted with systemd as init system (PID 1). Can't operate.
+ systemctl enable monoserve.service
+ '[' false == true ']'
+ systemctl enable onlyofficeRadicale.service
+ systemctl enable onlyofficeTelegram.service
+ systemctl enable onlyofficeSocketIO.service
+ systemctl enable onlyofficeThumb.service
+ systemctl enable onlyofficeFeed.service
+ systemctl enable onlyofficeIndex.service
+ systemctl enable onlyofficeJabber.service
+ systemctl enable onlyofficeMailAggregator.service
+ systemctl enable onlyofficeMailWatchdog.service
+ systemctl enable onlyofficeMailCleaner.service
+ systemctl enable onlyofficeNotify.service
+ systemctl enable onlyofficeBackup.service
+ systemctl enable onlyofficeStorageMigrate.service
+ systemctl enable onlyofficeStorageEncryption.service
+ systemctl enable onlyofficeUrlShortener.service
+ '[' SERVER == SERVER ']'
+ mv /app/config/nginx/prepare-onlyoffice /etc/nginx/sites-enabled/onlyoffice
+ service nginx stop
+ systemctl enable nginx.service
Synchronizing state of nginx.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable nginx
++ ps auxf
++ grep cron
++ grep -v grep
++ awk '{print $2}'
+ PID=
+ '[' ']'
+ systemctl enable elasticsearch.service
Synchronizing state of elasticsearch.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable elasticsearch
+ '[' -n '' ']'
+ '[' true == true ']'
+ exec /lib/systemd/systemd
Failed to mount tmpfs at /run: Operation not permitted
Failed to mount tmpfs at /run/lock: Operation not permitted
Failed to mount cgroup at /sys/fs/cgroup/systemd: Operation not permitted
[!!!!!!] Failed to mount API filesystems, freezing.
Freezing execution.
```

**Comment by Carazyda** [comment]
Hello @yoyecros Please download a new script for the installation, see [instruction](https://helpcenter.onlyoffice.com/installation/workspace-install-docker.aspx).

**Comment by yoyecros** [comment]
Hello @Carazyda 
I'm using the docker version, not a script.
I see that the image has been updated a day ago, is there any instruction before updating to avoid this error?
Thanks

**Comment by SuberSeb** [comment]
Update from 10 to 11 was failed for me too.

**Comment by Carazyda** [comment]
In close time we will update instructions for docker. For now we recommend use our script.

**Comment by aaronouthier** [comment]
Having the same issue. I filed an issue in the proper place (docker-communityserver).

**Comment by Carazyda** [comment]
Need to add one more mount to docker run command. In close time we will update a github instruction.
`-v /sys/fs/cgroup:/sys/fs/cgroup:ro`

**Comment by yoyecros** [comment]
Not working here.

Logs:
```
+ sed '/web\.talk/s/value=\"\S*\"/value=\"true\"/g' -i /var/www/onlyoffice/WebStudio/web.appsettings.config
,+ sed '/web\.chat/s/value=\"\S*\"/value=\"true\"/g' -i /var/www/onlyoffice/WebStudio/web.appsettings.config
,+ systemctl stop onlyofficeRadicale
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeTelegram
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeSocketIO
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeThumb
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeFeed
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeIndex
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeJabber
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeMailAggregator
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeMailWatchdog
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeMailCleaner
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeNotify
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeBackup
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeStorageMigrate
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeStorageEncryption
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop onlyofficeUrlShortener
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop god
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl enable god
,+ systemctl stop elasticsearch
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop redis-server
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop mysql
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop nginx
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl stop monoserveApiSystem.service
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl enable monoserveApiSystem.service
,++ seq 1 1
,+ for serverID in $(seq 1 ${APP_MONOSERVE_COUNT})
,+ index=1
,+ '[' 1 == 1 ']'
,+ index=
,+ systemctl stop monoserve.service
,System has not been booted with systemd as init system (PID 1). Can't operate.
,+ systemctl enable monoserve.service
,+ '[' false == true ']'
,+ systemctl enable onlyofficeRadicale.service
,+ systemctl enable onlyofficeTelegram.service
,+ systemctl enable onlyofficeSocketIO.service
,+ systemctl enable onlyofficeThumb.service
,+ systemctl enable onlyofficeFeed.service
,+ systemctl enable onlyofficeIndex.service
,+ systemctl enable onlyofficeJabber.service
,+ systemctl enable onlyofficeMailAggregator.service
,+ systemctl enable onlyofficeMailWatchdog.service
,+ systemctl enable onlyofficeMailCleaner.service
,+ systemctl enable onlyofficeNotify.service
,+ systemctl enable onlyofficeBackup.service
,+ systemctl enable onlyofficeStorageMigrate.service
,+ systemctl enable onlyofficeStorageEncryption.service
,+ systemctl enable onlyofficeUrlShortener.service
,+ '[' SERVER == SERVER ']'
,+ mv /app/config/nginx/prepare-onlyoffice /etc/nginx/sites-enabled/onlyoffice
,+ service nginx stop
,+ systemctl enable nginx.service
,Synchronizing state of nginx.service with SysV service script with /lib/systemd/systemd-sysv-install.
,Executing: /lib/systemd/systemd-sysv-install enable nginx
,++ ps auxf
,++ grep cron
,++ awk '{print $2}'
,++ grep -v grep
,+ PID=
,+ '[' ']'
,+ systemctl enable elasticsearch.service
,Synchronizing state of elasticsearch.service with SysV service script with /lib/systemd/systemd-sysv-install.
,Executing: /lib/systemd/systemd-sysv-install enable elasticsearch
,Created symlink /etc/systemd/system/multi-user.target.wants/elasticsearch.service → /usr/lib/systemd/system/elasticsearch.service.
,+ '[' -n '' ']'
,+ '[' true == true ']'
,+ exec /lib/systemd/systemd
,Failed to set up the root directory for shared mount propagation: Operation not permitted
,systemd 237 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid)
,Detected architecture x86-64.
,
,Welcome to Ubuntu 18.04.5 LTS!
,
,Set hostname to <1aefa83c6864>.
,Cannot determine cgroup we are running in: No medium found
,Failed to allocate manager object: No medium found
,[!!!!!!] Failed to allocate manager object, freezing.
,Freezing execution.
,
```

**Comment by aaronouthier** [comment]
@yoyecros I managed to pull an all-nighter last night and fixed the problem on my system.
First, you need 2 read/write persistent volumes of type tmpfs .

```
cd /volume/with/docker-communityserver
mkdir run lock
sudo mount -t tmpfs -o nusuid,noexec tmpfs run
sudo mount -t tmpfs -o nosuid,noexec tmpfs lock
```
Note: if you are already the user root, omit the word ‘sudo’ at the beginning of the last 2 lines.

Then add the following 4 lines to docker-compose.yml, in the volumes section of community server:

```
     - /sys/fs/cgroup:/sys/fs/cgroup:ro
     - /sys/fs/cgroup/systemd:/sys/fs/cgroup/systemd:ro
     - ./run:/run
     - ./lock:/run/lock
```
 Or, if you aren’t using docker-compose, change the 4 lines into 1 line and add a ‘v’ after each ‘-‘, such as:

-v /sys/fs/cgroup:/sys/fs/cgroup:ro -v /sys/fs/cgroup/systemd:/sys/fs/cgroup/systemd:ro
(Add the 2 other lines at the end of the one above.

**Comment by yoyecros** [comment]
@aaronouthier thanks for your help!
Here is my docker-compose file (only volume part):
```
volumes:
     - /app/onlyoffice/CommunityServer/data:/var/www/onlyoffice/Data
     - /app/onlyoffice/CommunityServer/logs:/var/log/onlyoffice
     - /app/onlyoffice/CommunityServer/DocumentServerData:/var/www/onlyoffice/DocumentServerData
     - /app/onlyoffice/CommunityServer/data/certs:/var/www/onlyoffice/Data/certs
     - /sys/fs/cgroup:/sys/fs/cgroup:ro
     - /sys/fs/cgroup/systemd:/sys/fs/cgroup/systemd:ro
     - /app/onlyoffice/CommunityServer/run:/run
     - /app/onlyoffice/CommunityServer/lock:/run/lock
```

I have done the mount commands for `/app/onlyoffice/CommunityServer/run` and `/app/onlyoffice/CommunityServer/lock` before starting the container. Did you see something wrong?

**Comment by aaronouthier** [comment]
This is what I have:

```
> volumes:
>      - community_data:/var/www/onlyoffice/Data
>      - community_log:/var/log/onlyoffice
>      - document_data:/var/www/onlyoffice/DocumentServerData
>      - ./certs:/var/www/onlyoffice/Data/certs
>      - /sys/fs/cgroup:/sys/fs/cgroup:ro
>      - /sys/fs/cgroup/systemd:/sys/fs/cgroup/systemd:ro
>      - ./run:/run
>      - ./lock:/run/lock
```

From there, type: ‘docker-compose up -d’ . This is obvious for those familiar with docker-compose, and not to those who aren’t. ☺️

If you have already done this, then I am curious if it worked?

**Comment by aaronouthier** [comment]
It just occurred to me, I did make one other change to my compose file, but I’m not sure if it was needed. I added an export for port 8000 in document server. I’m still learning docker stuff. As I said, I’m not sure if it was needed. I just noticed in the logs that community server was waiting for a connection to document server on port 8000, and added the export. This was before I figured out what was causing the community server to fail/freeze. I never did test if this was also an issue. It ain’t broke for me, so I didn’t try to fix it...

**Comment by aaronouthier** [comment]
Also, in order to correctly run docker-compose, there has to be the docker-compose.yml in the current working directory.

**Comment by aaronouthier** [comment]
Also, on a side note: the docker version of community server has it’s own section on GitHub. This issue is currently filed in the non-docker section.

**Comment by aaronouthier** [comment]
Please see my bug report here:
https://github.com/ONLYOFFICE/Docker-CommunityServer/issues/104

**Comment by yoyecros** [comment]
I'm working with docker-compose. I added the 8000 port but same result.
But finally I made this working by removing document and comunity containers, then docker-compose up -d. Now I can access my portal. Thanks for your informations :)

I don't know if it's necessary to add all elements (4 volumes, mount run and lock, and 8000 port), but I don't really want to break again my config for testing...

Thanks again :)

**Comment by SuberSeb** [comment]
> I'm working with docker-compose. I added the 8000 port but same result.
> But finally I made this working by removing document and comunity containers, then docker-compose up -d. Now I can access my portal. Thanks for your informations :)
> 
> I don't know if it's necessary to add all elements (4 volumes, mount run and lock, and 8000 port), but I don't really want to break again my config for testing...
> 
> Thanks again :)

Can you show your docker-compose please?

**Comment by aaronouthier** [comment]
@SuberSeb 
https://github.com/ONLYOFFICE/Docker-CommunityServer/issues/104#issuecomment-713513618

**Comment by aaronouthier** [comment]
Remember to change the mail hostname.
